### DISCUSSION TITLE
How are KnowledgeBase entity_vectors used?

### DISCUSSION THREAD

**Discussion by kinghuang** [explosion/spaCy#9095] [Help: Model Advice] [answered]
When creating a [KnowledgeBase](https://spacy.io/api/kb), how does the [`entity_vector`](https://spacy.io/api/kb#add_entity) impact a KnowledgeBase or EntityLinker's performance? It's not clear to me if there's some sort of strategy involved in the choice of entity vectors.

In the [nel_emerson tutorial](https://github.com/explosion/projects/tree/v3/tutorials/nel_emerson), the entity vectors are derived from the description of the entities (e.g., *Australian tennis player*) rather than the entity's name (e.g., *Roy Stanley Emerson*). Would it be incorrect to use the name or id for the vector instead of the description? Or, what if there are two Australian tennis players with a last name "Emerson". I assume it wouldn't work to use the same entity vectors for both, based on the description?

**Comment by svlandeg** [comment, marked as answer]
In a nutshell: the entity vectors are compared to sentence embeddings, and the more similar an entity vector is to a sentence embedding, the more likely the Entity Linker will deem the link between the mention in that sentence, and the ID corresponding to the entity (vector). During training, the Entity Linker learns an embedding model that minimizes the distance between sentence embeddings and entity vectors of gold-standard links.

If you have two people called "Emerson" and you'd give them the same entity vector, the EL will effectively not be able to distinguish between the two. It might still create correct predictions based on the aliases that you add for each to the KB, because the EL works in two steps:

- generate plausible candidates from the KB, given the mention from text (this depends on the aliases of the entities)
- define the candidate with the highest similarity between sentence encoding & entity vector - i.e. disambiguate according to the context in which the mention appears

Does that clarify things?

**Reply by kinghuang** [reply]
Yes, that helps! I've also been wondering about the sentencizer's role in EntityLinker, and that explains it, too. Thanks!
